src/main/storm/mesos/MesosNimbus.java

```diff
@@ -383,461 +635,269 @@ public void assignSlots(Topologies topologies, Map<String, Collection<WorkerSlot
+        }
+        List<Resource> workerCpuResources = getResourcesScalar(offerResources, workerCpu, "cpus");
+        offerResources = subtractResourcesScalar(offerResources, workerCpu, "cpus");
+        List<Resource> workerMemResources = getResourcesScalar(offerResources, workerMem, "mem");
+        offerResources = subtractResourcesScalar(offerResources, workerMem, "mem");
+        List<Resource> workerPortsResources = getResourcesRange(offerResources, slot.getPort(), "ports");
+        offerResources = subtractResourcesRange(offerResources, slot.getPort(), "ports");
+
+        // Find port for the logviewer
+        if (!subtractedExecutorResources && MesosCommon.startLogViewer(_conf)) {
+          List<Integer> portList = new ArrayList<>();
+          collectPorts(offerResources, portList, 1);
+          int port = Optional.fromNullable((Number) _conf.get(Config.LOGVIEWER_PORT)).or(8000).intValue();
+          executorPortsResources = getResourcesRange(offerResources, port, "ports");
+          if (!executorPortsResources.isEmpty()) {
+            // Was the port available?
+            extraConfig = " -c " + MesosCommon.AUTO_START_LOGVIEWER_CONF + "=true";
+            offerResources = subtractResourcesRange(offerResources, port, "ports");
+          }
+        }
+        Offer remainingOffer = existingBuilder.addAllResources(offerResources).build();
 
-  public class NimbusScheduler implements Scheduler {
-    private CountDownLatch _registeredLatch = new CountDownLatch(1);
+        // Update the remaining offer list
+        offers.put(id, remainingOffer);
 
-    public void waitUntilRegistered() throws InterruptedException {
-        _registeredLatch.await();
-    }
+        String configUri;
+        try {
+          configUri = new URL(_configUrl.toURL(),
+              _configUrl.getPath() + "/storm.yaml").toString();
+        } catch (MalformedURLException e) {
+          throw new RuntimeException(e);
+        }
 
-    @Override
-    public void registered(final SchedulerDriver driver, FrameworkID id, MasterInfo masterInfo) {
-      _driver = driver;
-      try {
-        _state.put(FRAMEWORK_ID, id.getValue());
-      } catch (IOException e) {
-        LOG.error("Halting process...", e);
-        Runtime.getRuntime().halt(1);
-      }
-      _offers = new RotatingMap<OfferID, Offer>(
-          new RotatingMap.ExpiredCallback<OfferID, Offer>() {
-            @Override
-            public void expire(OfferID key, Offer val) {
-              driver.declineOffer(val.getId());
-            }
+        String executorDataStr = JSONValue.toJSONString(executorData);
+        ExecutorInfo.Builder executorInfoBuilder = ExecutorInfo.newBuilder();
+        executorInfoBuilder
+            .setExecutorId(ExecutorID.newBuilder().setValue(details.getId()))
+            .setData(ByteString.copyFromUtf8(executorDataStr));
+        if (!subtractedExecutorResources) {
```

Comment:
This is the logic causing [the problem I described here](https://github.com/mesos/storm/pull/65#issuecomment-161906189).  I'm sending a PR in a second.